vuu-ui/packages/vuu-layout/src/layout-persistence/LocalLayoutPersistenceManager.ts

```
@@ -8,72 +8,124 @@ const metadataSaveLocation = "layouts/metadata";
+          resolve(layouts[0].json);
+        })
+        .catch(e => reject(e));
+    });
   }
 
-  loadMetadata(): LayoutMetadata[] {
-    return getLocalEntity<LayoutMetadata[]>(metadataSaveLocation) || [];
+  loadMetadata(): Promise<LayoutMetadata[]> {
+    return new Promise((resolve) => {
+      const metadata = getLocalEntity<LayoutMetadata[]>(metadataSaveLocation);
+      resolve(metadata || []);
+    })
   }
 
-  private loadLayouts(): Layout[] {
-    return getLocalEntity<Layout[]>(layoutsSaveLocation) || [];
+  private loadLayouts(): Promise<Layout[]> {
+    return new Promise((resolve) => {
+      const layouts = getLocalEntity<Layout[]>(layoutsSaveLocation);
+      resolve(layouts || []);
+    });
   }
 
-  private appendAndPersist(newId: string,
-                           newMetadata: Omit<LayoutMetadata, "id">,
-                           newLayout: LayoutJSON,
-                           existingLayouts: Layout[],
-                           existingMetadata: LayoutMetadata[]) {
-    existingLayouts.push({id: newId, json: newLayout});
-    existingMetadata.push({id: newId, ...newMetadata});
+  private appendAndPersist(
+    newId: string,
+    newMetadata: Omit<LayoutMetadata, "id">,
+    newLayout: LayoutJSON,
+    existingLayouts: Layout[],
+    existingMetadata: LayoutMetadata[]
+  ) {
+    existingLayouts.push({ id: newId, json: newLayout });
+    existingMetadata.push({ id: newId, ...newMetadata });
 
     this.saveLayoutsWithMetadata(existingLayouts, existingMetadata);
   }
 
-  private saveLayoutsWithMetadata(layouts: Layout[], metadata: LayoutMetadata[]): void {
+  private saveLayoutsWithMetadata(
+    layouts: Layout[],
+    metadata: LayoutMetadata[]
+  ): void {
     saveLocalEntity<Layout[]>(layoutsSaveLocation, layouts);
     saveLocalEntity<LayoutMetadata[]>(metadataSaveLocation, metadata);
   }
+
+  private async validateIds(id: string): Promise<void> {
+    return Promise
+      .all([
```

could we have just one .catch() on the promise returned by Promise.all() rather than 2 on the individual promises?

---

We can, but it would only report one error (`Promise.all` fails fast, so would jump to the `catch` as soon as one error comes back). I went with this approach to provide a complete error, but we can use a single catch if performance is more of a concern?